theia-ide/theia
# [DRAFT] use task definition as a reference to compare tasks

- Although task users have much flexibility to define tasks through
contributing to the Task definitions, TaskConfiguration.equals() and
ContributedTaskConfiguration.equals() compare certain properties of task
configs, and thus are not reliable enough. With changes in this pull
request, Theia uses task definitions as a reference to decide which
properties in task configs should be considered in comparison.
- fixed #5878

Signed-off-by: Liang Huang <liang.huang@ericsson.com>

#### How to test

**This change can only be tested after https://github.com/theia-ide/theia/pull/5915 gets merged.**
That's why I marked it as [DRAFT]. 

![Peek 2019-08-18 00-47](https://user-images.githubusercontent.com/37082801/63220284-f648f900-c151-11e9-803d-cd8dc114dce4.gif)

- Use npm extension
- Configure one or more npm tasks, and refresh page
- Go to Terminal -> Run Task menu to check if the configured task(s) are displayed correctly
- Check if the configured task(s) can be run


#### Review checklist
